homeassistant/components/tasmota/binary_sensor.py

```diff
@@ -0,0 +1,91 @@
+"""Support for Tasmota binary sensors."""
+import logging
+
+from homeassistant.components import binary_sensor
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+import homeassistant.helpers.event as evt
+
+from .const import DOMAIN as TASMOTA_DOMAIN
+from .discovery import TASMOTA_DISCOVERY_ENTITY_NEW, clear_discovery_hash
+from .mixins import TasmotaAvailability, TasmotaDiscoveryUpdate
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up Tasmota binary sensor dynamically through discovery."""
+
+    @callback
+    def async_discover(tasmota_entity, discovery_hash):
+        """Discover and add a Tasmota binary sensor."""
+        try:
+            async_add_entities(
+                [
+                    TasmotaBinarySensor(
+                        tasmota_entity=tasmota_entity, discovery_hash=discovery_hash
+                    )
+                ]
+            )
+        except Exception:
```

Comment:
What exception can happen here?

Reply:
None, will open a new PR to clean that up.